packages/i18n/src/translateWithMarkup.tsx

```diff
@@ -52,18 +58,22 @@ const VOID_ELEMENTS = [
  * @param {?Object.<string, ReactElement>} mapping Map of tag names to React components.
  * @return {ReactNode} Transformed node.
  */
-export function transformNode(node, mapping = {}) {
-  const { childNodes, localName, nodeType, textContent } = node;
+export function transformNode(
+  node: ChildNode,
+  mapping: Mapping = {}
+): React.ReactNode {
+  const { firstChild, nodeType, textContent } = node;
   if (Node.TEXT_NODE === nodeType) {
     return textContent;
   }
 
-  const children = node.hasChildNodes()
-    ? [...childNodes].map((child) => transform(child, mapping))
-    : null;
+  const children = node.hasChildNodes() ? transform(firstChild, mapping) : null;
 
-  if (localName in mapping) {
-    return cloneElement(mapping[localName], null, children);
+  if ('localName' in node) {
+    const { localName } = node as Element;
+    if (localName in mapping) {
+      return cloneElement(mapping[localName], undefined, children);
```

Comment:
For the life of me, I can't figure out how to get TS to accept the type of this one? I tried everything, but I can't match the types to [any of the overloads available](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L292-L324). And similarly for `createElement` and [its overloads](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L254-L288) a few lines below.